### PR TITLE
feat(perf): improve performance of conversation filter API [CW-1605]

### PR DESCRIPTION
**Pull request by scmmishra**
## Description

This PR fixes some N+1 queries for the conversation filter service. This caches the labels and precomputes the total message count. 

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

### Baseline

Total Queries: 251

CleanShot 2023-05-18 at 18 07 44@2x

### Optimized

Total Queries: 213

CleanShot 2023-05-18 at 18 08 15@2x

### CODE REVIEW

**Comment by scmmishra** [comment]
@sojan-official @pranavrajs If you see the following lines https://github.com/chatwoot/chatwoot/blob/590ce788b99c7634936af07bf999c05abadb0416/app/views/api/v1/conversations/partials/_conversation.json.jbuilder#L23-L25

We request the `push_event_data` from the messages, this includes attachments too.

https://github.com/chatwoot/chatwoot/blob/590ce788b99c7634936af07bf999c05abadb0416/app/models/message.rb#L134-L148

We can remove the attachments from the query with an optional parameter `include_attachments` and use it in the jbuilder, this will reduce the number of queries further

**Comment by scmmishra** [comment]
@sojan-official @pranavrajs found another opportunity,

https://github.com/chatwoot/chatwoot/blob/590ce788b99c7634936af07bf999c05abadb0416/app/models/contact.rb#L109-L121

The `avatar_url` is causing a N+1 query, I tried adding `{ messages: { sender: { avatar_attachment: [:blob] }]}}` to the `include` in `filter_service.rb` but it didn't help. LMK if you have any ideas.

Regardless, the current PR should also provide some improvements

File: app/views/api/v1/conversations/partials/_conversation.json.jbuilder [hunk 1]
```diff
@@ -17,7 +17,7 @@ json.meta do
 end
 
 json.id conversation.display_id
-if conversation.messages.count.zero?
+if conversation.messages.first.blank?
   json.messages []
 elsif conversation.unread_incoming_messages.count.zero?
```
Comment (sojan-official) [inline app/views/api/v1/conversations/partials/_conversation.json.jbuilder:22]: we can avoid the count query here.  do first.present?

Comment (scmmishra) [reply app/views/api/v1/conversations/partials/_conversation.json.jbuilder:22]: I did `first.present?` earlier, then rubocop suggested to make it `first.blank?`Are the two different in their query execution?

Comment (sojan-official) [reply app/views/api/v1/conversations/partials/_conversation.json.jbuilder:22]: `first.blank?` should be good . lets do the same for  line:22 as well.  the count query seems overkill there as well

**Review by sojan-official — COMMENTED** [review]
LGTM. let's take the  other suggestions in subsequent PRs

File: app/views/api/v1/conversations/partials/_conversation.json.jbuilder (hunk 1 above)
Comment (scmmishra) [reply app/views/api/v1/conversations/partials/_conversation.json.jbuilder:22]: We could do that but then `conversation.unread_incoming_messages.count` is fetched again on line 43, unconditionally. So I figured it better be fetched on 22 as well

**Review by sojan-official — APPROVED** [review]

**Pull request merged**